signing/src/main/java/tech/pegasys/web3signer/signing/config/metadata/parser/YamlMapperProvider.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2023 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.signing.config.metadata.parser;
+
+import static com.fasterxml.jackson.databind.MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Optional;
+
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+import org.yaml.snakeyaml.LoaderOptions;
+
+public enum YamlMapperProvider {
```

Comment:
nit: I like the enum approach but is there any advantage to it if we still need to use a synchronized init method to achieve lazy init?

Reply:
I don't like the enum approach. This is not really an enum it's just using it for instantiation and doesn't hide the initialisation that you usually have with a singleton.

Reply:
I've converted enum to standard class with singelton pattern.